src/ExpressReceiver.ts

```diff
@@ -134,7 +134,7 @@ export default class ExpressReceiver extends EventEmitter implements Receiver {
     return new Promise((resolve, reject) => {
       // TODO: what about synchronous errors?
       this.server.close((error) => {
-        if (error) {
+        if (Boolean(error)) {
```

Comment:
whoa, def do not want to be boxing these values in objects. since `error` is typed to `Error | undefined`, i think its better to change this condition to `error !== undefined`.

Reply:
yeah, definitely right. surprised a test didn't catch this somewhere?

Reply:
I agree `error !== undefined` is better. Out of curiosity, isn't this what the code was doing before, just implicitly?

Reply:
i learned something new today! when i quickly read over this before, i thought we were boxing boolean primitives in Boolean objects, which behave in confusingly different ways. in actuality, this was `Boolean` used _as a function_, which does something different.

I found a small hint about this in the [MDN docs for logical operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Double_NOT_!!):

> The same conversion can be done through the Boolean function.

Double NOT is essentially coercing a value to a boolean primitive using the "truthiness" rules in JavaScript. the original code here was intended to evaluate the truthiness of `error`, so the Boolean function makes a lot of sense.

The [tslint rule which makes the original code fail](https://palantir.github.io/tslint/rules/strict-boolean-expressions/) is on because depending on the truthiness is a common pitfall in JavaScript, because its not immediately clear which values are and are not going to evaluate to the "true" branch. But I don't see how using the Double NOT (or the Boolean function) is helpful in making it more clear.

I think we have to decide if we find value in that rule or not. If we want to keep it, then we should explicitly check for the values that we care about instead of using truthiness. If we don't want to keep it, then using the Boolean function seems like a reasonable choice.

There's also a number of options to the tslint rule to consider. The most relevant one to us would be `allow-undefined-union`, which if we turned on should allow `Error | undefined` in conditionals without an error.

Thoughts?

Reply:
I think it ended up catching some interesting cases here, and led to better code, so probably worth keeping?  Avoiding checking truthiness is probably for the best.